src/ModbusSlave.cpp

```diff
@@ -147,6 +147,10 @@ int Modbus::poll() {
     // check minimum length.
     if (lengthIn < 8) return 0;
 
+    // this removes the null character that sometimes
+    // is get from the SoftwareSerial on the "nano"
+    if(lengthIn > 8 && bufIn[lengthIn-1] == 0) lengthIn--;
```

Comment:
This is the reason why I added the "lengthIn > 8" control ...

Reply:
I'm confused :-)

none valid message of length 11 [CRC is 9 10]:
1 2 3 4 5 6 7 8 9 10 0
valid message of length 10 [CRC is 9 0] :
1 2 3 4 5 6 7 8 9 0

both pass the test of len > 8 and lastByte == 0 ? what am I missing ?

Reply:
p.s. 
FC1,2,3,4,5 all length 8 but we also support FC16 with length 13

Reply:
ok, got it, you are right :)
I got the problem in the FC_READ_INPUT_REGISTERS case:
```
if (lengthIn != 8) return 0;
```
If you don't know in advance the length of the message, you may want to change the check, based on the final CRC instead of the buffer length.

Reply:
👍